src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -246,11 +246,36 @@ class BaseDropdown extends Component<DropdownProps> {
     this.setState({ showPopover: false });
   }
 
+  private handleSpaceAndEnter = (
+    popoverItems: Array<ReactElement<DropdownItemProps>>,
+    getNextItem: () => ReactElement<DropdownItemProps>,
+  ) => {
+    const { focusedDescendantId, showPopover } = this.state;
+    if (!showPopover) {
+      this.openPopover();
+      if (!focusedDescendantId) {
+        const nextItem = getNextItem();
+        if (nextItem) {
+          this.setState({ focusedDescendantId: nextItem.props.value });
+        }
+      }
+    } else if (showPopover && focusedDescendantId) {
+      const focusedItem = popoverItems.find(
+        (item) => item?.props.value === focusedDescendantId,
+      );
+      if (focusedItem && !focusedItem.props.disabled) {
+        this.handleItemSelection(focusedItem.props.value);
+      }
+    }
+  };
+
   private handleKeyDown = (event: React.KeyboardEvent) => {
     const { focusedDescendantId, showPopover } = this.state;
     const popoverItems = Array.isArray(this.props.children)
       ? this.props.children
-      : [this.props.children];
+      : this.props.children !== undefined
```

Comment:
Changed logic here, because previously it was possible to have popOverItems = [undefined] (truthy) and this never returned